lib/services/profiles-api-service.js

```diff
@@ -102,41 +102,80 @@ function profileApiServiceFactory(
 
                 });
             } else {
-                return self.createNodeAndRunDiscovery(macAddresses);
+                return self.createNodeAndRunDiscovery(macAddresses, options);
             }
         });
     };
 
-    ProfileApiService.prototype.runDiscovery = function(node) {
+    ProfileApiService.prototype.runDiscovery = function(node, options) {
         var self = this;
+        var configuration;
 
+        if (node.type === 'switch') {
+            configuration = self.getSwitchDiscoveryConfiguration(node, options.switchVendor);
+        } else {
+            configuration = {
+                name: 'Graph.SKU.Discovery',
+                options: {
+                    defaults: {
+                        graphOptions: {
+                            target: node.id
+                        },
+                        nodeId: node.id
+                    }
+                }
+            };
+        }
+
+        // The nested workflow holds the lock against the nodeId in this case,
+        // so don't add it as a target to the outer workflow context
+        return workflowApiService.createAndRunGraph(configuration, null)
+        .then(function() {
+            return self.waitForDiscoveryStart(node.id);
+        })
+        .then(function() {
+            return node;
+        });
+    };
+
+    ProfileApiService.prototype.getSwitchDiscoveryConfiguration = function(node, vendor) {
         var configuration = {
-            name: 'Graph.SKU.Discovery',
+            name: 'Graph.SKU.Switch.Discovery.Active',
             options: {
                 defaults: {
                     graphOptions: {
                         target: node.id
                     },
                     nodeId: node.id
+                },
+                'vendor-discovery-graph': {
+                    graphName: null
                 }
             }
         };
 
-        return workflowApiService.createAndRunGraph(configuration)
-        .then(function() {
-            return self.waitForDiscoveryStart(node.id);
-        })
-        .then(function() {
-            return node;
-        });
+        vendor = vendor.toLowerCase();
+
+        if (vendor === 'cisco') {
```

Comment:
Would it be better to call vendor specific workflows over hard coding vendor types in support of a generic workflow definition?  In other words, could 'Graph.Switch.Discovery.Cisco.Poap' be invoked directly based on the profile set for specific vendors (perhaps based on MAC prefix or vendor ID)?